File: java/src/main/java/com/iteso/nintendo/Rayquaza.java
package com.iteso.nintendo;

import com.iteso.nintendo.Moves.*;

/**
 * Created by rvillalobos on 2/24/18.
 */
public class Rayquaza extends PokemonCharacter {
    /**
     * Hit Points are the maximum life of pokemon.
     */
    public static final int HIT_POINTS = 100;
    /**
     * Defense multiplier value between 0-1.
     */
    public static final double DEFENSE_MULTIPLIER = 0.4;
    /**
     * Damage from 1 - 20.
     */
    public static final int MAIN_ATTACK_DAMAGE = 5;
    /**
     * Damage from 1 -25.
     */
    public static final int SECOND_ATTACK_DAMAGE = 15;

    /**
     * Rayquaza constructor.
     */
    public Rayquaza() {
        DragonBreath dragonBreath = new DragonBreath();
        DragonClaw dragonClaw = new DragonClaw();
        setType("air");
        setName("Rayquaza");
        setHasEvolution(true);
        setSecondAttack(dragonBreath);
        setMainAttack(dragonClaw);
        setHitPoints(HIT_POINTS);
        setDefenseMultiplier(DEFENSE_MULTIPLIER);
        setMainAttackDamage(dragonBreath.attackDamage());
        setSecondAttackDamage(dragonClaw.attackDamage());
    }

    @Override
    public final String evolve() {
        return null;
    }

    @Override
    public final String defend(final int attack) {
        int damage;

        damage = (int) (attack * getDefenseMultiplier());
        int newHP = getHitPoints() - damage;

        String defendMessage = new String("Defending attack, damage caused is "
                + damage + " new HP is " + newHP);

        setHitPoints(newHP);
        return defendMessage;

    }

    @Override
    public final String secondAttack() {

        String attackMessage = new String("Attacking opponent with "
                + getSecondAttack()
                + " causing a damage of " + getSecondAttackDamage());
        return attackMessage;


    }

    @Override
    public final String mainAttack() {
        String attackMessage = new String("Attacking opponent with "
                + getMainAttack()
                + " causing a damage of " + getMainAttackDamage());
        return attackMessage;

    }

    @Override
    public void setNewAttack(int attack, int attackDamage, Attacks newAttack) {
        if (attack == 1) {
            setMainAttack(newAttack);
            setMainAttackDamage(attackDamage);
        } else {
            setSecondAttack(newAttack);
            setSecondAttackDamage(attackDamage);
        }
    }


    @Override
    public String attackName() {
        return null;
    }

    @Override
    public void setAttackName(String attackName) {

    }

    @Override
    public int attackDamage() {
        return 0;
    }

    @Override
    public void setAttackDamage(int attackDamage) {

    }
}

File: python/src/main/python/Rayquaza.py
from PokemonCharacter import *
from attacks.Dragonbreath import *
from attacks.Dragonclaw import *

class Rayquaza(PokemonCharacter):
    def __init__(self):
        self.HIT_POINTS = 75
        self.DEFENSE_MULTIPLIER = 0.4
        self.dragonbreath = Dragonbreath()
        self.dragonclaw = Dragonclaw()
        self.setType('dragon')
        self.setName('Rayquaza')
        self.setHasEvolution(False)
        self.setMainAttack(self.dragonbreath)
        self.setSecondAttack(self.dragonclaw)
        self.setHitPoints(self.HIT_POINTS)
        self.setDeffenseMultiplier(self.DEFENSE_MULTIPLIER)
        self.setMainAttackDamage(self.dragonbreath.getAttackDamage())
        self.setSecondAttackDamage(self.dragonclaw.getAttackDamage())

    def evolve(self):
        return None

    def defend(self, attack):
        damage = attack * self.getDefenseMultiplier()
        newHP = self.getHitPoints() - damage
        defendMessage = 'Defending attack, damage caused is '
        defendMessage += str(int(damage))
        defendMessage += ' new HP is '
        defendMessage += str(int(newHP))
        self.setHitPoints(int(newHP))
        return defendMessage

    def performSecondAttack(self):
        attackMessage = 'Attacking opponent with '
        attackMessage += self.getSecondAttack()
        attackMessage += ' causing a damage of '
        attackMessage += str(self.getSecondAttackDamage())
        return attackMessage

    def performMainAttack(self):
        attackMessage = 'Attacking opponent with '
        attackMessage += self.getMainAttack()
        attackMessage += ' causing a damage of '
        attackMessage += str(self.getMainAttackDamage())
        return attackMessage

    def setNewAttack(self, attack, attackDamage, newAttack):
        if(attack == 1):
            self.setMainAttack(newAttack)
            self.setMainAttackDamage(attackDamage)
        else:
            self.setSecondAttack(newAttack)
            self.setSecondAttackDamage(attackDamage)
